lib/bindings/HTTPBinding.js

```diff
@@ -122,8 +122,8 @@ function executeCommand(apiKey, device, serializedPayload, callback) {
     }
 
     request(options, function(error, response, body) {
-        if (error || (response.statusCode !== 200 && response.statusCode !== 201)) {
-            callback(new errors.HTTPCommandResponseError(response.statusCode, error));
+        if (error || (response && response.statusCode !== 200 && response.statusCode !== 201)) {
+            callback(new errors.HTTPCommandResponseError(response ? response.statusCode : 400, error));
```

Comment:
I think the logic should be error || !response ||(response.statusCode !== 200 && response.statusCode !== 201) - for the current predicate, the *400* clause will never fire. because if `resource` is *undefined* the `else` clause will run.

Reply:
fixed in https://github.com/telefonicaid/iotagent-json/pull/454/commits/773c6884f46ee039ab5166497c7acff932f50baa